StandaloneMmPkg/Include/Protocol/PiMmCpuDriverEp.h

```diff
@@ -0,0 +1,54 @@
+/** @file
+  This file describes the PiMm Cpu Driver Entry Point Protocol.
+  the protocol is for defining handler for mm communication request event
+  according to cpu driver.
+
+  Copyright (c) 2024, Arm Limited. All rights reserved.<BR>
+  SPDX-License-Identifier: BSD-2-Clause-Patent
+
+**/
+
+#ifndef PI_MM_CPU_DRIVER_EP_H_
+#define PI_MM_CPU_DRIVER_EP_H_
+
+#include <PiMm.h>
+
+#define PI_MM_CPU_DRIVER_EP_GUID  { \
+  0x6ecbd5a1, 0xc0f8, 0x4702, { 0x83, 0x01, 0x4f, 0xc2, 0xc5, 0x47, 0x0a, 0x51 } \
+  }
+
+/**
+  The PI Standalone MM entry point for handling mm communication request
+  according to cpu driver.
+  At first, mm communication request is delivered via smc to DelegatedEventLoop.
+  In DelegatedEventLoop, after SMC call sanity checking, it calls this function
+  to further handling via MmiHandler based on mm communication protocol which
+  saved in CommBufferAddr.
+
+  @param  [in] EventId            The event Id based on firmware.
+  @param  [in] CpuNumber          The CPU number.
+  @param  [in] CommBufferAddr     Address of the communication buffer.
+
+  @retval   EFI_SUCCESS             Success.
+  @retval   EFI_INVALID_PARAMETER   A parameter was invalid.
+  @retval   EFI_ACCESS_DENIED       Access not permitted.
+  @retval   EFI_OUT_OF_RESOURCES    Out of resources.
+  @retval   EFI_UNSUPPORTED         Operation not supported.
+**/
+typedef
+EFI_STATUS
+(EFIAPI *PI_MM_CPU_DRIVER_ENTRYPOINT)(
+  IN UINTN  EventId,
+  IN UINTN  CpuNumber,
+  IN UINTN  CommBufferAddr
+  );
+
+typedef struct _EDKII_PI_MM_CPU_DRIVER_EP_PROTOCOL EDKII_PI_MM_CPU_DRIVER_EP_PROTOCOL;
+
+struct _EDKII_PI_MM_CPU_DRIVER_EP_PROTOCOL {
+  PI_MM_CPU_DRIVER_ENTRYPOINT    PiMmCpuDriverEntryPoint;
+};

```

Comment:
X86 does not need such protocol for StandaloneMmCoreEntryPointLib.DelegateEventLoop() to find the CPU driver's entry point.
As CPU driver's entry point in X86 is directly saved in a CPU register.

So, how about renaming this protocol to be ARM specific? e.g.: EDKII_ARM_MM_CPU_DRIVER_EP_PROTOCOL?

Reply:
IIRC, RiscV uses CPU driver (but its usage isn't upstreamed):
    commit e7a71694467a (" StandaloneMmPkg: Make StandaloneMmCpu driver architecture independent")

So I think it would be better to sustain this naming convention.